### PR TITLE
Setup abstractions

### PR DESCRIPTION
**Pull request by JakovTomasic**
Solves #6

### CODE REVIEW

File: client/src/types/Locale.ts
Comment (Marko132001) [inline client/src/types/Locale.ts:1]: I think we aren't gonna need this class to implement locales. If needed we can create it later.

File: client/src/types/Theme.ts
Comment (Marko132001) [inline client/src/types/Theme.ts:1]: Same as Locale class, we can add this if needed later.

Comment (JakovTomasic) [reply client/src/types/Theme.ts:1]: The `Theme` class was added just for defining the settings interface (client/src/logic/interfaces/Settings.ts). If I should remove this class then I should remove the Theme getter and setter in the interface, too?

Theme (if not removed) should be a wrapper class with data important to support themes (which may depend on the library used). The one who implements the feature will define what data is it.

For example, if we implement a theme toggle like [the Toggle Button here](https://dev.to/chinmaymhatre/implementing-dark-mode-in-nextjs-with-tailwind-css-and-next-themes-a4e#:~:text=Implementing%20Dark%20Mode-,Next.,theme%20is%20set%20to%20dark.) then we will use `next-themes` library. I treat this library as an external dependency that should be easily changed for another one. By extracting that into the settings interface it's completely decoupled from the rest of the code. You can see on the given link that the button also has logic for handling system and user themes which I think is a technical dept (a button should just define UI interface. No app-wide logic). This is more obvious if we need to store the last set theme in cookies. The button component would need to read and write cookies? I think that's weird.

I agree that this approach may be too complicated for an application of this size and that introducing said technical depts would not be much of a problem.

After more clearly explaining the reasoning behind the class, do you still think we should remove it (together with the part of the settings interface)? I'm good with either decision.

File: client/src/types/Locale.ts
Comment (JakovTomasic) [reply client/src/types/Locale.ts:1]: Same reasoning as for the Theme class. Should I remove the class?

File: client/src/types/Theme.ts
Comment (Marko132001) [reply client/src/types/Theme.ts:1]: Thanks for the explanation, I didn't really understand the purpose at first. It makes more sense when you put it like that. My idea was as you said to implement the logic inside the toggle button component (due to very small amount of code needed for implementation). Nevermind that now, we can leave it as it is.

File: client/src/types/Locale.ts
Comment (Marko132001) [reply client/src/types/Locale.ts:1]: Leave it as it is.

**Review by Marko132001 — APPROVED** [review]
All looks good to me. You can merge this branch to main.

**Pull request merged**